pkg/nvml/device.go

```diff
@@ -2931,10 +2931,56 @@ func (l *library) DeviceGetRunningProcessDetailList(device Device) (ProcessDetai
 }
 
 func (device nvmlDevice) GetRunningProcessDetailList() (ProcessDetailList, Return) {
+	return deviceGetRunningProcessDetailList(device)
+}
+
+func deviceGetRunningProcessDetailList(device nvmlDevice) (ProcessDetailList, Return) {
 	var plist ProcessDetailList
 	plist.Version = STRUCT_VERSION(plist, 1)
-	ret := nvmlDeviceGetRunningProcessDetailList(device, &plist)
-	return plist, ret
+	plist.NumProcArrayEntries = 1
+
```

Comment:
Why is it not sufficient to just do this:
```
var plist ProcessDetailList
	plist.Version = STRUCT_VERSION(plist, 1)
	plist.NumProcArrayEntries = 1
	plist.ProcArray = &(make([]ProcessDetail_v1, plist.NumProcArrayEntries)[0])
	...
```

Does NVML realloc the space that `plist.ProcArray` points to inside the call?

Reply:
Actually, I just looked it up -- it says this:
> To determine the size of the plist->procArray array to allocate, call the function with plist->numProcArrayEntries set to zero and plist->procArray set to NULL. The return code will be either NVML_ERROR_INSUFFICIENT_SIZE (if there are valid processes of type plist->mode to report on, in which case the plist->numProcArrayEntries field will indicate the required number of entries in the array) or NVML_SUCCESS (if no processes of type plist->mode exist).

Which means we should be following the same pattern as e.g.:
https://github.com/NVIDIA/go-nvml/blob/main/pkg/nvml/device.go#L993

Reply:
With no need to introduce all of this c-level malloc wrapper, etc.

Reply:
Have you run it locally? I did encounter a runtime error.